docs(tree): Replace usages of the term checkout with view (#15889)
microsoft/FluidFramework: experimental/dds/tree2/README.md

See [indexes and branches](./docs/indexes%20and%20branches.md) for details on how this works with branches.

> When applications want access to the `tree`'s data, they do so through a [`SharedTreeView`](./src/shared-tree/sharedTreeView.ts) which abstracts the indexes into nice application facing APIs.

Reviewer: ```suggestion
When applications want access to the `tree`'s data, they do so through an [`ISharedTreeView`](./src/shared-tree/sharedTreeView.ts) which abstracts the indexes into nice application facing APIs.
```

Views may also have state from the application, including:

-   [`view-schema`](./src/core/schema-view/README.md)
-   adapters for out-of-schema data
-   request or hints for what subsets of the tree to keep in memory
-   pending transactions
-   registrations for application callbacks / events.

[`shared-tree`](./src/shared-tree/) provides a default view which it owns, but applications can create more if desired, which they will own.
Since views subscribe to events from `shared-tree`, explicitly disposing any additionally created ones of is required to avoid leaks.

> [transactions](./src/core/transaction/README.md) are created from `SharedTreeView`s and are currently synchronous.

Reviewer: ```suggestion
[transactions](./src/core/transaction/README.md) are created by `ISharedTreeView`s and are currently synchronous.
```

    subgraph "@fluid-experimental/tree2"
        shared-tree--"configures"-->shared-tree-core
        shared-tree-core--"Summary"-->Indexes--"Summary"-->ForestIndex;

>         ForestIndex--"Exposed by"-->SharedTreeView

Reviewer: ```suggestion
        ForestIndex--"Exposed by"-->ISharedTreeView
```

    end

>     SharedTreeView--"viewed by"-->app

Reviewer: ```suggestion
    ISharedTreeView--"viewed by"-->app
```

```

[`shared-tree`](./src/shared-tree/) configures [`shared-tree-core`](./src/shared-tree-core/README.md) with a set of indexes.
`shared-tree-core` downloads the summary data from the Fluid Container, feeding the summary data (and any future edits) into the indexes.

> `shared-tree` then constructs the default `SharedTreeView`.

Reviewer: ```suggestion
`shared-tree` then constructs the default view.
```

        transaction--"collects edits in"-->EditBuilder
        EditBuilder--"updates anchors"-->AnchorSet
        EditBuilder--"deltas for edits"-->transaction

>         transaction--"applies deltas to"-->forest["SharedTreeView's forest"]

Reviewer: ```suggestion
        transaction--"applies deltas to"-->forest["ISharedTreeView's forest"]
```
